components/layout_thread_2020/lib.rs

```diff
@@ -734,6 +734,7 @@ impl LayoutThread {
         let rayon_pool = STYLE_THREAD_POOL.lock().unwrap();
         let rayon_pool = rayon_pool.pool();
         let rayon_pool = rayon_pool.as_ref();
+        let rayon_pool = rayon_pool.filter(|p| p.current_thread_index().is_some());
```

Comment:
Does this fix need to be extended for legacy layout as well?

So I'm still a bit confused how this works in Gecko. Does `current_thread_index()` actually return a value there for the main thread? If so, why? This approach seems okay, though we would be removing the main thread from the thread pool AFAIU. It would be nice to keep consistency with Gecko allowing the main thread to participate, as that was done for performance reasons on embedded devices, as I understand.

In either case, I think it might be nice to put a comment here explaining the situation as otherwise this is quite mysterious.

Reply:
Yes I forgot, legacy needs it too

Reply:
Fixed upstream: https://github.com/servo/stylo/commit/0c3139b534f8a950eb283261b820e7636d665639